### PR TITLE
MODBULKOPS-287 - Add name fields to User etnity

### PR DESCRIPTION
**Pull request by AntonAntonich**
[MODBULOPS-287](https://folio-org.atlassian.net/browse/MODBULKOPS-287)

## Purpose
Expand User entity with middleName and preferredFirstName

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [x] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by AntonAntonich** [comment]
image

File: src/main/java/org/folio/bulkops/mapper/UserToUserDtoMapper.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package org.folio.bulkops.mapper;
+
+import org.folio.bulkops.domain.bean.User;
+
+public interface UserToUserDtoMapper {
+  org.folio.bulkops.domain.dto.User mapUserToUserDto(User user);
```
Comment (khandramai) [inline src/main/java/org/folio/bulkops/mapper/UserToUserDtoMapper.java:6]: Is it possible to add package org.folio.bulkops.domain.dto.User to imports?

File: src/main/java/org/folio/bulkops/mapper/UserToUserDtoMapperImpl.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+@Component
+public class UserToUserDtoMapperImpl implements UserToUserDtoMapper {
+
+  @Override
+  public org.folio.bulkops.domain.dto.User mapUserToUserDto(User user) {
+    Personal userPersonal = user.getPersonal();
+    return new org.folio.bulkops.domain.dto.User()
+      .id(UUID.fromString(user.getId()))
+      .firstName(user.getPersonal().getFirstName())
```
[the remark this answers is not in the document]
Comment (AntonAntonich) [reply src/main/java/org/folio/bulkops/mapper/UserToUserDtoMapperImpl.java]: fixed

File: src/main/java/org/folio/bulkops/mapper/UserToUserDtoMapper.java (hunk 1 above)
Comment (AntonAntonich) [reply src/main/java/org/folio/bulkops/mapper/UserToUserDtoMapper.java:6]: Two classes with the same name, one of them should contain full class name

File: src/main/java/org/folio/bulkops/mapper/UserToUserDtoMapperImpl.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package org.folio.bulkops.mapper;
+
+import java.util.UUID;
+import org.folio.bulkops.domain.bean.Personal;
+import org.folio.bulkops.domain.bean.User;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserToUserDtoMapperImpl implements UserToUserDtoMapper {
```
Comment (khandramai) [inline src/main/java/org/folio/bulkops/mapper/UserToUserDtoMapperImpl.java:9]: org.folio.bulkops.domain.bean.User - entity for csv-file processing (with whole data), org.folio.bulkops.domain.dto.User - entity with a limited set of fields to show information about user who executed bulk operations. So, this looks like a mix of two user entities from absolutely different business flows. As a result this mapper introduces confusion into the code. Is it possible to get rid of it and transform the fields as was done originally?

File: src/main/java/org/folio/bulkops/mapper/UserToUserDtoMapper.java (hunk 1 above)
Comment (khandramai) [reply src/main/java/org/folio/bulkops/mapper/UserToUserDtoMapper.java:6]: https://github.com/folio-org/mod-bulk-operations/pull/211/files#r1623905907

File: src/main/java/org/folio/bulkops/mapper/UserToUserDtoMapperImpl.java (hunk 3 above)
Comment (AntonAntonich) [reply src/main/java/org/folio/bulkops/mapper/UserToUserDtoMapperImpl.java]: done

**Review by obozhko-folio — APPROVED** [review]

**Review by alekGbuz — APPROVED** [review]

**Pull request merged**